### DISCUSSION TITLE
What the recommended way to send audio to deepgram?

### DISCUSSION THREAD

**Discussion by nlinx** [deepgram/community#753] [General help] [answered]
I'm sending 3s chunks of audio to Deepgram that are associated with each other. i.e. if I say a sentence it may be split into 3 or 4 audio chunks. Right now I'm sending it all in and concating the results from the stream to get the full sentence transcription. Is there a better way?

Also, I'm finding that the speech_final field is somewhat unreliable. I've been tweaking the endpoint configuration when instantiating the client to try to get it to work but it's not very consistent. Most of the time I'll have to send in silent audio to get it to return speech_final = true. Is there a better way to work with this field?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
Hi @nlinx 

I would recommend not NOT concatenating distinct sections of audio. You should just be able to send the audio segments are they are to get transcription on them.

For "flushing" the remaining audio bits, take a look at this post here:
https://github.com/orgs/deepgram/discussions/769#discussioncomment-9500121
